src/compositemark.ts

```diff
@@ -72,3 +79,175 @@ add(ERRORBAR, (spec: GenericUnitSpec<ERRORBAR, Encoding>): LayerSpec => {
     ]
   };
 });
+
+add(BOX, (spec: GenericUnitSpec<BOX, Encoding>): LayerSpec => {
+  const {mark: _m, encoding: encoding, ...outerSpec} = spec;
+  const {x: _x, y: _y, ...nonPositionEncoding} = encoding;
+
+  let fullEncoding: any = {};
```

Comment:
This name is kind of weird since it's not really the `fullEncoding`. It's rather a key-value list of different fieldDefs you want to reuse.  
I'd say get rid of it and instead declare a variable for each encoding mixins (not fieldDef) that you *need* to re-use.  

In fact, with [this suggestion](https://github.com/vega/vega-lite/pull/2037/files#r107746245), you won't really need that many variables pre-declared.  

For the ones that you still need, avoid using `let` if you can reasonably use `const`.  
(`let` is okay sometimes, but whenever `const` seems reasonable -- never use `let`.)